src/Agent.Worker/JobRunner.cs

```diff
@@ -34,8 +36,14 @@ public sealed class JobRunner : AgentService, IJobRunner
             }
 
             // Setup the job server and job server queue.
+            var vssConnection = ApiUtil.GetVssConnection(message);
+
             var jobServer = HostContext.GetService<IJobServer>();
-            await jobServer.ConnectAsync(ApiUtil.GetVssConnection(message));
+            await jobServer.ConnectAsync(vssConnection);
+
+            var agentServer = HostContext.GetService<IAgentServer>();
+            await agentServer.ConnectAsync(vssConnection);
```

Comment:
Agree. We should have a separate server class. Also, we cannot use vssConnection returned by GetVssConnection to initialize agentServer as it would be pointing to RMO in case of release.